kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/V3ResourcesModule.java

```diff
@@ -30,5 +39,38 @@ protected void configure() {
     bindAsContract(ChunkedOutputFactory.class);
     bindAsContract(StreamingResponseFactory.class);
     bind(Clock.systemUTC()).to(Clock.class);
+    bindFactory(ProduceResponseExecutorServiceFactory.class)
+        .to(ExecutorService.class)
+        .in(Singleton.class);
+  }
+
+  private static final class ProduceResponseExecutorServiceFactory
+      implements Factory<ExecutorService> {
+
+    private final int produceResponseThreadPoolSize;
+
+    @Inject
+    ProduceResponseExecutorServiceFactory(
+        @ProduceResponseThreadPoolSizeConfig Integer produceExecutorThreadPoolSize) {
+      this.produceResponseThreadPoolSize = produceExecutorThreadPoolSize;
+    }
+
+    @Override
+    public ExecutorService provide() {
+      ThreadFactory namedThreadFactory =
+          new ThreadFactoryBuilder().setNameFormat("Produce-response-thread-%d").build();
+      return new ThreadPoolExecutor(
+          produceResponseThreadPoolSize,
+          produceResponseThreadPoolSize,
+          0L,
+          TimeUnit.MILLISECONDS,
+          new LinkedBlockingQueue<Runnable>(),
+          namedThreadFactory);
+    }
```

Comment:
nit: I actually think we should use this ctor, but with the core pool size being `Runtime.getRuntime().availableProcessors()` and the max pool size being the configurable size, defaulting to `Runtime.getRuntime().availableProcessors()` as well. If you prefer not to go for that, which I think is OK, because we will be fine tuning this code soon, you can just use `Executors.newFixedThreadPool(produceResponseThreadPoolSize, namedThreadFactory)`.

Reply:
I would make everything configurable, and using `Runtime.getRuntime().availableProcessors()` as a reasonable default.

Reply:
The default is set to Runtime.getRuntime().availableProcessors() in KafkaRestConfig, so hopefully that covers the default setting, and then it's configurable just via the normal properties.

Dimitar - thanks for pointing out that method, I looked and looked for it but obviously my eyes weren't working, so I made my own based on what Executors.newFixedThreadPool() does.  I've flipped back to using that.